#!/bin/sh /usr/share/dpatch/dpatch-run
## 13-improve-makefile.dpatch by Michael Karcher
## <debian@mkarcher.dialup.fu-berlin.de>
##
## DP: Makes CFLAGS overridable and removes executable on clean

@DPATCH@
diff -urNad rott-1.0+dfsg~/Makefile rott-1.0+dfsg/Makefile
--- rott-1.0+dfsg~/Makefile	2008-01-10 09:18:13.000000000 +0100
+++ rott-1.0+dfsg/Makefile	2008-01-10 09:18:14.000000000 +0100
@@ -47,9 +47,13 @@
 
 
 CC = gcc
-CFLAGS = -g $(SDL_CFLAGS) -DUSE_SDL=1 -DPLATFORM_UNIX=1 -W -Wall -Wno-unused $(EXTRACFLAGS) -DDATADIR=\"/usr/share/games/rott/\"
+CFLAGS = -g -O2 -W -Wall -Wno-unused
+ALL_CFLAGS = $(SDL_CFLAGS) -DUSE_SDL=1 -DPLATFORM_UNIX=1 $(EXTRACFLAGS) -DDATADIR=\"/usr/share/games/rott/\" $(CFLAGS)
 LDLIBS = $(SDL_LDFLAGS) -lSDL -lSDL_mixer $(EXTRALDFLAGS) -Wl,-E
 
+%.o: %.c
+	$(CC) $(ALL_CFLAGS) -c $< -o $@
+
 all: rott
 
 rott: 	\
@@ -105,7 +109,8 @@
 	$(CC) $^ $(LDLIBS) -o $@
 
 clean:
-	rm -rf *.o
+	rm -f *.o
+	rm -f rott
 
 distclean: clean
-	rm -rf *~
+	rm -f *~
